griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -0,0 +1,170 @@
+import uuid
+from typing import Optional
+from attr import define, field
+from dataclasses import dataclass
+from griptape.drivers import BaseVectorStoreDriver
+from sqlalchemy.engine import Engine
+from sqlalchemy import create_engine, Column, String, JSON
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Session
+from pgvector.sqlalchemy import Vector
+
+
+def vector_model_factory(table_name="griptape_vectors"):
```

Comment:
Can you move this inside `PgVectorVectorStoreDriver`? Griptape convention is to keep all static methods as class methods.

Reply:
Can you also add type hints for `table_name` and the return type?

Reply:
Do we need a default here?